src/libsync/syncengine.cpp

```diff
@@ -91,7 +91,12 @@ SyncEngine::SyncEngine(AccountPtr account, const QString& localPath,
     url_string = Utility::toCSyncScheme(url_string);
 
     csync_create(&_csync_ctx, localPath.toUtf8().data(), url_string.toUtf8().data());
-    csync_init(_csync_ctx);
+
+    const QString dbFile = this->journalDbFilePath();
+    _journal->setDatabaseFilePath( dbFile );
```

Comment:
Why should the sync engine set this ?   The journal is created already and might have been opened before.
IMHO, the creator of the journal should set this (i.e: the folder)